__test__/integration/loan_offer/runners/accept.ts

```diff
@@ -0,0 +1,95 @@
+import * as Web3 from "web3";
+
+import { BigNumber, Dharma } from "../../../../src";
+
+import { LoanOffer } from "../../../../src/types";
+
+import { ContractsAPI } from "../../../../src/apis/";
+
+import { DummyTokenContract } from "../../../../src/wrappers";
+
+// Accounts
+import { ACCOUNTS } from "../../../accounts";
+
+const { Token } = Dharma.Types;
+
+const CREDITOR = ACCOUNTS[0];
+const DEBTOR = ACCOUNTS[1];
+
+const TX_DEFAULTS = { from: CREDITOR.address, gas: 4712388 };
+
+export async function testAccept(
+    dharma: Dharma,
+    params: any,
+    signAndAccept: (loanOffer: LoanOffer, address: string) => void,
+) {
+    describe("passing valid params", () => {
+        let loanOffer: LoanOffer;
+
+        let web3: Web3;
+
+        let contractsApi: ContractsAPI;
+
+        let principalToken: DummyTokenContract;
+        let collateralToken: DummyTokenContract;
+
+        beforeAll(async () => {
+            loanOffer = await LoanOffer.createAndSignAsCreditor(dharma, params);
+
+            const provider = new Web3.providers.HttpProvider("http://localhost:8545");
+            web3 = new Web3(provider);
+
+            contractsApi = new ContractsAPI(web3);
+
+            principalToken = await DummyTokenContract.at(
+                (await contractsApi.loadTokenBySymbolAsync(params.principalToken)).address,
+                web3,
+                TX_DEFAULTS,
+            );
+
+            collateralToken = await DummyTokenContract.at(
+                (await contractsApi.loadTokenBySymbolAsync(params.collateralToken)).address,
+                web3,
+                TX_DEFAULTS,
+            );
+        });
+
+        beforeEach(async () => {
+            // set balances
+            await principalToken.setBalance.sendTransactionAsync(
```

Comment:
similar to what we do in `loan`, let's create a function that sets these balances and allowances for us.

```export async function setBalancesAndAllowances(
    dharma: Dharma,
    params: LoanRequestParams,
    debtor: string,
    creditor: string,
): Promise<void> {
    // Configure balances and allowances for debtor.
    await setBalanceForSymbol(dharma, params.collateralAmount, params.collateralToken, debtor);
    await setUnlimitedAllowanceForSymbol(dharma, params.collateralToken, debtor);

    // Configure balances and allowances for creditor.
    await setBalanceForSymbol(dharma, params.principalAmount, params.principalToken, creditor);
    await setUnlimitedAllowanceForSymbol(dharma, params.principalToken, creditor);
}
```

Reply:
done.